homeassistant/components/emulated_hue/__init__.py

```diff
@@ -280,35 +288,32 @@ def is_entity_exposed(self, entity):
             # Ignore entities that are views
             return False
 
-        domain = entity.domain.lower()
-        explicit_expose = entity.attributes.get(ATTR_EMULATED_HUE, None)
+        if entity.entity_id in self._entities_with_hidden_attr_in_config:
+            return not self._entities_with_hidden_attr_in_config[entity.entity_id]
+
         explicit_hidden = entity.attributes.get(ATTR_EMULATED_HUE_HIDDEN, None)
+        explicit_expose = entity.attributes.get(ATTR_EMULATED_HUE, None)
 
-        if (
-            entity.entity_id in self.entities
-            and CONF_ENTITY_HIDDEN in self.entities[entity.entity_id]
-        ):
-            explicit_hidden = self.entities[entity.entity_id][CONF_ENTITY_HIDDEN]
+        if explicit_expose is not None:
+            get_deprecated(
```

Comment:
What does this even do? It doesn't return anything. It just prints a message? And how long has it been printing it ? Can we remove it ?

Reply:
commit fd9ceb73811f6698afe60318728e6cdc0dfba0cf
Author: rbflurry <rbflurry@gmail.com>
Date:   Tue Sep 26 03:31:35 2017 -0400

I'd say kill with fire

Reply:
Will remove and update the tests

Reply:
The docs actually say `emulated_hue_hidden` is going away, and it appears `emulated_hue` has already been removed from the docs

```
source/_docs/configuration/customizing-devices.markdown:emulated_hue_hidden:
source/_docs/configuration/customizing-devices.markdown:  description: Set to `true` to hide the entity from `emulated_hue` (this will be deprecated in the near future and should be configured in [`emulated_hue`](/integrations/emulated_hue)).
source/_docs/configuration/customizing-devices.markdown:      emulated_hue_hidden: false
```

Reply:
commit d691f95103174876128f108f018c3fc73e412f92
Author: DubhAd <DubhAd@users.noreply.github.com>
Date:   Fri Jun 29 21:07:36 2018 +0100

```
    Added details about emulated_hue_hidden (#5631)
    
    Given that it mentions in the emulated_hue pages that `emulated_hue_hidden` is being deprecated, it seems sensible to actually mention that here ;)```

Reply:
`emulated_hue` was changed to prefer `emulated_hue_hidden` in 0.55

commit 58a48dbf9a3cd6e10787a0b4e9b4cc62c61c83ad
Author: Ken Davidson <ken@ragerobotics.com>
Date:   Mon Oct 9 02:03:02 2017 -0400

    Updated "emulated_hue_hidden" setting to match change in 0.55. (#3565)

Reply:
@balloob 
I'm removing `emulated_hue` attribute since it was swapped to `emulated_hue_hidden` in 0.55
I'm tempted to remove `emulated_hue_hidden` since it has been documented as deprecated for 20 months

Reply:
Kill it all

Reply:
Gladly :)